pymc_experimental/bart/bart.py

```diff
@@ -40,12 +40,18 @@ def _shape_from_params(self, dist_params, rep_param_idx=1, param_shapes=None):
         )
 
     def _infer_shape(cls, size, dist_params, param_shapes=None):
-        dist_shape = (cls.X.shape[0],)
+        if cls.shape is not None:
```

Comment:
Another question, why not make use of the aesara size parameter, instead of having a self.shape?

Reply:
Not sure what I have to change. I tried using size instead of shape and I got 

ValueError: Tensor of type TensorType(int64, (0,)) could not be cast to have 0 dimensions

Reply:
I don't mean here in specific, I mean in the whole BartRV. It looked like size is never used, and `self.shape` was basically doing what size does in vanilla RVs.

Reply:
For instance if you support `size` then you don't even need this method. You can just override `_supp_shape_from_params`,

Which just has to return `cls.X.shape[0]`. Aesara will then figure out how size expands from this core dimension automatically whenever.

Reply:
Yeah, I got that. But my attempt to use size instead of shape was not successful :-( I could try again tomorrow.